### PR TITLE
Sitemap should only include `page` routes

### LINKED ISSUE #7080: @astrojs/sitemap 1.3.1 adds endpoints with trailing slash
### What version of `astro` are you using?

2.4.5

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Linux

### What browser are you using?

Edge

### Describe the Bug

Starting with astrojs/sitemap 1.3.0, endpoints like `rss.xml` are included in `sitemap-0.xml`.
Should sitemap include endpoints (data, images, rss etc.) at all?

If so, the trailing slash needs to be trimmed as the URLs are invalid.
`<url><loc>https://example.com/rss.xml/</loc></url>`

Repro: Run build in stackblitz sample and inspect `sitemap-0.xml`

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-xfdxg4

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

This is a MAJOR (breaking) change that will bump `@astrojs/sitemap` to 2.0.0!

- Ensures that `@astrojs/sitemap` only includes page routes
- Fix #7080. See that issue for more details. Our documentation refers to the `sitemap` as generating a sitemap from your **pages**, but all routes were erroneously included. This behavior did not match user expectations and non-page routes needed to be filtered out manually.

## Testing

Our tests only covered pages, I added an endpoint and redirect to the suite.

## Docs

This change aligns with our existing docs for this package

### CODE REVIEW

**Review by matthewp — APPROVED** [review]

**Comment by natemoo-re** [comment]
Flagging this for @withastro/maintainers-docs just in case there are concerns!

**Review by bluwy — APPROVED** [review]

**Review by ematipico — APPROVED** [review]

**Review by sarah11918 — APPROVED** [review]
Thanks for checking in with docs @natemoo-re !

Reminder: this PR does not update the README (which may be fine!)

Here is how the README introduces the `sitemap` integration. Are you still comfortable with this description? (I'll point out that it does NOT strictly say *page* routes in the very first line, even though that may be the expectation.)

I'm happy to sign off and approve here, and will leave it to you to confirm that you still believe the README intro presents this integration accurately. If not, then update as you see fit!  :raised_hands:

**Pull request merged**
